modules/efk-logging-fluentd-undefined.adoc

```diff
@@ -0,0 +1,121 @@
+|CDM_EXTRA_KEEP_FIELDS
+|Specify an extra set of defined fields to be kept in addition to the `CDM_DEFAULT_KEEP_FIELDS`. The default is "".
+|CDM_EXTRA_KEEP_FIELDS="broker"
+
+|CDM_KEEP_EMPTY_FIELDS
+|Specify defined fields to retain even if empty. Empty defined fields not specified are dropped. The default is "".
+|CDM_KEEP_EMPTY_FIELDS="offset"
+
+|CDM_USE_UNDEFINED
+|Set to `true` to move undefined fields to the `undefined` top level field. The default is `false`. If `true`, values in `CDM_DEFAULT_KEEP_FIELDS`
+and `CDM_EXTRA_KEEP_FIELDS` are not moved to `undefined`.
+|CDM_USE_UNDEFINED=true
+
+|CDM_UNDEFINED_NAME
+|Specify a name for the undefined top level field if using `CDM_USE_UNDEFINED`. The default is`undefined`.
+Enabled only when `CDM_USE_UNDEFINED` is `true`.
+|CDM_UNDEFINED_NAME="undef"
+
+|CDM_UNDEFINED_MAX_NUM_FIELDS
+|If the number of undefined fields is greater than this number, all undefined fields are converted to their JSON string representation 
+and stored in the `CDM_UNDEFINED_NAME` field. If the record contains more than this value of undefined fields, no further processing takes place on these fields. 
+Instead, the fields will be converted to a single string JSON value, stored in the top-level `CDM_UNDEFINED_NAME` field. Keeping the default of `-1`allows for an unlimited number of undefined fields, which is not recommended.
+Enabled only when `CDM_USE_UNDEFINED` is `true`.
+|CDM_UNDEFINED_MAX_NUM_FIELDS=4
+
+|CDM_UNDEFINED_TO_STRING
+|Set to `true` to convert all undefined fields to their JSON string representation. The default is `false`.
+|CDM_UNDEFINED_TO_STRING=true
+
+|CDM_UNDEFINED_DOT_REPLACE_CHAR
+|Specify a character to use in place of a dot character '.' in an undefined field. `MERGE_JSON_LOG` must be `true`. The default is `UNUSED`. 
+If you set the MERGE_JSON_LOG parameter to `true`, see the xref:mergejson[Note below].
+|CDM_UNDEFINED_DOT_REPLACE_CHAR="_"
+
+|===
+
+[[mergejson]]
+[NOTE]
+====
+If you set the `MERGE_JSON_LOG` parameter in the log collector daemonset and CDM_UNDEFINED_TO_STRING environment variables to true, you might receive an Elasticsearch *400* error. 
+The error occurs because when`MERGE_JSON_LOG=true`, the log collector adds fields with data types other than string. When you set CDM_UNDEFINED_TO_STRING=true, 
+the log collector attempts to add those fields as a string value resulting in the Elasticsearch 400 error. The error clears when the log collector rolls over 
+the indices for the next day’s logs
+
+When the log collector rolls over the indices, it creates a brand new index. The field definitions are updated and you will not get the *400* error.
+====
+
+.Procedure
+
+Use the `CDM_*` parameters to configure undefined and empty field processing.
+
+. Configure how to process defined fields, as needed:
+
+.. Specify the defined fields to move using the `CDM_DEFAULT_KEEP_FIELDS` and `CDM_EXTRA_KEEP_FIELDS`.
+
+.. Set `CDM_KEEP_EMPTY_FIELDS` to `true` to retain empty defined fields.
```

Comment:
Set `CDM_KEEP_EMPTY_FIELDS` to `true` may cause some misunderstanding here, because the value of `CDM_KEEP_EMPTY_FIELDS` is in the CSV format(the same type as the `CDM_DEFAULT_KEEP_FIELDS` and `CDM_EXTRA_KEEP_FIELDS`), and the user may think `true` is a boolean value.

For example:
CDM_KEEP_EMPTY_FIELDS="loggerName"